PeraWallet/Scenes/SwapV2/SwapViewController.swift

```diff
@@ -384,7 +385,24 @@ final class SwapViewController: BaseViewController {
                 bannerController?.presentErrorBanner(title: "title-error", message: .empty)
                 return
             }
-            swapAssetFlowCoordinator.getQuote(account: selectedAccount, assetIn: assetIn, assetOut: assetOut, amount: value)
+            swapAssetFlowCoordinator.getQuote(account: selectedAccount, assetIn: assetIn, assetOut: assetOut, amount: value, slippage: sharedViewModel?.slippageSelected.map { Decimal(floatLiteral: $0.value) })
+        case .calculatePeraFee(forAmount: let amount, withPercentage: let percentage):
+            guard let assetIn = selectedAssetIn?.asset else { return }
+            swapAssetFlowCoordinator.onFeeCalculated = { [weak self] peraFee, error in
+                guard let self = self else { return }
+                if let error {
+                    bannerController?.presentErrorBanner(title: "title-error", message: error.prettyDescription)
+                    return
+                }
+                
+                let fee = peraFee?.fee?.assetAmount(fromFraction: assetIn.decimals) ?? 0
+                let swapAmount = calculateSwapAmount(balance: amount.toDecimal, fee: fee, percentage: percentage.toDecimal)
+                
+                sharedViewModel?.payingText = Formatter.decimalFormatter(minimumFractionDigits: 0, maximumFractionDigits: 8).string(for: swapAmount) ?? SwapSharedViewModel.defaultAmountValue
+                sharedViewModel?.isLoadingPayAmount = false
+            }
+           
+            swapAssetFlowCoordinator.calculateFee(assetIn: assetIn, amount: amount)
```

Comment:
This case have quite complex logic. Consider moving it to the function to make it more readable.